### PR TITLE
Update SHA-384/512 Blocksize + Test

### PR DESCRIPTION
**Pull request by jtmcdole**
When creating Sha512; copypasta of a variable that was only used in the HMAC process would lead to an error. Update that and add test vectors for all HMAC/SHA-2 functions.

### CODE REVIEW

**Comment by jtmcdole** [comment]
@lrhn or @jonasfj  - this one should be (hopefully?) uncontroversial.

**Comment by postmasters** [comment]
Woot! Thanks, @jtmcdole !

File: pubspec.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: crypto
-version: 2.1.1+1
+version: 2.1.1+2
```
Comment (jonasfj) [inline pubspec.yaml]: Bug fixes should be a patch version, not a build version.

Comment (kevmoo) [reply pubspec.yaml]: Also include a changelog entry?

Also, behavior change should cause `2.1.2`

Comment (postmasters) [reply pubspec.yaml]: This definitely requires a bump in version. The old behavior causes HMAC-SHA512 to produce wrong values. So it's best to let people know that they should use 2.1.2 from now on.

**Comment by jtmcdole** [comment]
I'll bump the version / update docs when I get home tonight.

**Comment by jonasfj** [comment]
Published: https://pub.dev/packages/crypto/versions/2.1.2

**Pull request merged**